### PR TITLE
Fix clang-tidy warnings

### PR DESCRIPTION
**Pull request by cyyever**
### Description

This PR fixes more clang-tidy warnings and adds more checks.
### Motivation and Context

Better code.

### CODE REVIEW

File: onnx/defs/schema.h [hunk 1]
```diff
@@ -1381,7 +1381,10 @@ class OpSchemaRegistry final : public ISchemaRegistry {
   class OpSchemaRegisterOnce final {
    public:
     // Export to cpp custom register macro
-    OpSchemaRegisterOnce(OpSchema op_schema, int opset_version_to_load = 0, bool fail_duplicate_schema = true) {
+    explicit OpSchemaRegisterOnce(
```
Comment (justinchuby) [inline onnx/defs/schema.h:1384]: Does this change the api?

File: onnx/version_converter/helper.cc [hunk 2]
```diff
@@ -36,20 +36,15 @@ void assert_numpy_multibroadcastable(
...
-  const std::vector<Dimension>* A_ptr;
-  const std::vector<Dimension>* B_ptr;
-  int A;
-  int B;
+  const std::vector<Dimension>* A_ptr = &input1_sizes;
+  const std::vector<Dimension>* B_ptr = &input2_sizes;
+  int A = 1;
+  int B = 2;
   if (input1_sizes.size() < input2_sizes.size()) {
     A_ptr = &input2_sizes;
```
Comment (justinchuby) [inline onnx/version_converter/helper.cc:44]: Instead, declare in both branches ?

File: onnx/defs/data_type_utils.h [hunk 3]
```diff
@@ -52,7 +52,7 @@ class DataTypeUtils final {
  private:
   static void FromString(const std::string& type_str, TypeProto& type_proto);
 
-  static void FromDataTypeString(const std::string& type_str, int32_t& tensor_data_type);
+  static int32_t FromDataTypeString(const std::string& type_str);
```
Comment (justinchuby) [inline onnx/defs/data_type_utils.h:55]: Does this change the api?

**Review by justinchuby — COMMENTED** [review]
Lgtm as long as there is no api change. We need to be careful about changes in headers

File: onnx/defs/data_type_utils.h (hunk 3 above)
Comment (cyyever) [reply onnx/defs/data_type_utils.h:55]: Yes, but it is a private method

File: onnx/defs/schema.h (hunk 1 above)
Comment (cyyever) [reply onnx/defs/schema.h:1384]: Explicit in this case seems reasonable.

File: onnx/version_converter/helper.cc (hunk 2 above)
Comment (justinchuby) [reply onnx/version_converter/helper.cc:44]: The old behavior seems to declare first and initialize in the branches. If we declare in both branches we won’t have the risk of uninitialized variables this is solving. Please correct me if I am wrong

**Review by justinchuby — APPROVED** [review]

File: onnx/version_converter/helper.cc (hunk 2 above)
Comment (cyyever) [reply onnx/version_converter/helper.cc:44]: Can't declare in both branches because the variables are used below the branches.

**Review by justinchuby — APPROVED** [review]

File: onnx/defs/schema.h (hunk 1 above)
Comment (paveloom) [reply onnx/defs/schema.h:1384]: Just for reference, it broke [`onnxruntime`](https://github.com/microsoft/onnxruntime). See https://github.com/microsoft/onnxruntime/issues/24561 and https://github.com/onnx/onnx/issues/6708.

**Pull request merged**